pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -108,12 +119,59 @@ func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.Syn
 	uniqueRevisions := []int32{}
 	for _, nodeStatus := range operatorStatus.NodeStatuses {
 		revision := nodeStatus.CurrentRevision
+
+		// during day-0: pre-empt
+		if revision == 0 {
+			_, err = c.updateStatus(
+				ctx,
+				false, false,
+				reasonClusterInit,
+				"one or more static pod revision(s) are unknown",
+			)
+			return err
+		}
+
 		if !revisionMap[revision] {
 			revisionMap[revision] = true
 			uniqueRevisions = append(uniqueRevisions, revision)
 		}
 	}
 
+	// In case a checkProfileRejection func was used, status will be updated accordingly
+	// ("" profile is also default profile, so "" -> anotherProfile: rejection can be handled)
+	if c.checkProfileRejectionFn != nil {
+		isRejected, rejectMsg, err := c.checkProfileRejectionFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
+		if err != nil {
+			return err
+		}
+		// if profile transition is rejected, set status to progress=Unknown, complete=Unknown
+		// and degrade operator with suitable error message
+		if isRejected {
+			_, err = c.updateStatus(
+				ctx,
+				false, false, // not progressing, not complete
+				reasonLatencyProfileUpdateProhibited,
+				"",
+			)
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf(rejectMsg)
+		}
+	}
+
+	// in case profile was not rejected and,
+	// in case of empty workerlatency profile, set status Complete=False and pre-empt sync
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
```

Comment:
As discussed with @atiratree: There's a very low amount of risk involved in changing this. We might end up skipping a Progress status when user tries to go from Medium -> "", but that's the only risk. Apart from that at all times we could ideally expect the profile value to be empty only for clusters that do not use the worker latency profile feature at all, also since this is a non-GA feature having a no-op case with profile empty would make sense and we can change this behavior in the future before going GA.

Reply:
Added comment:
```
        // in case profile was not rejected and,
	// in case of empty workerlatency profile, set status Complete=False and pre-empt sync;
	// Note: There's a very low amount of risk involved with this, we might end up skipping a
	// Progress status when user tries to go from Medium -> "", but that's the only risk.
	// Apart from that at all times we could ideally expect the profile value to be empty only
	// for clusters that do not use the worker latency profile feature at all,
	// also since this is a non-GA feature having a no-op case with profile empty would make sense
	// and we can change this behavior in the future before Worker Latency Profiles feature is GA.
```